docs/website/docs/reference/explainers/how-dlt-works.md

```diff
@@ -6,33 +6,47 @@ keywords: [architecture, extract, normalize, load]
 
-## Normalize
+```py
+import dlt
 
-The configurable normalization engine in `dlt` recursively unpacks this nested structure into
-relational tables (i.e., inferring data types, linking tables to create nested relationships,
-etc.), making it ready to be loaded. This creates a
-[schema](../../general-usage/glossary.md#schema), which will automatically evolve to accommodate any future
-source data changes (e.g., new fields or tables).
+pipeline = dlt.pipeline(pipeline_name="my_pipeline", destination="duckdb")
+pipeline.run([{"id": 1}, {"id": 2}, {"id": 3}], table_name="items")
+```
 
-## Load
+This is what happens when the `run` method is executed:
 
-The data is then loaded into your chosen [destination](../../general-usage/glossary.md#destination).
-`dlt` uses configurable, idempotent, atomic loads that ensure data safely ends up there. For
-example, you don't need to worry about the size of the data you are loading, and if the process is
-interrupted, it is safe to retry without creating errors.
+1. [Extract](how-dlt-works.md#extract) - Fully extracts the data from your source to your hard drive. In the example above, an implicit source with one resource with 3 items is created and extracted.
+2. [Normalize](how-dlt-works.md#normalize) - Inspects and normalizes your data and computes a schema compatible with your destination. For the example above, the normalizer will detect one column `id` of type `int` in one table named `items` and no further normalization is needed.
+3. [Load](how-dlt-works#load) - Runs schema migrations if necessary on your destination and loads your data into the destination. For the example above, a new dataset on a local duckdb database is created that contains one table with one column `id` of type `int` and three rows of data.
+
+## The three phases
+
+### Extract
+
+During the extract phase, `dlt` fully extracts the data from your [sources](../../general-usage/source) to your hard drive into a new [load package](../../general-usage/destination-tables#load-packages-and-load-ids), which will be assigned a unique ID and will contain your raw data as received from your sources. Additionally, you can [supply schema hints](../../general-usage/source#define-schema) to define the data types of some of the columns or add a primary key and unique indexes. You can also control this phase by [limiting](../../general-usage/resource#sample-from-large-data) the number of items extracted in one run, using [incremental cursor fields](../../general-usage/incremental-loading#incremental-loading-with-a-cursor-field), and by tuning the performance with [parallelization](../../reference/performance#extract). You can also apply filters and maps to [obfuscate](../../general-usage/customising-pipelines/pseudonymizing_columns) or [remove](../../general-usage/customising-pipelines/removing_columns) personal data, and you can use [transformers](../../examples/transformers) to create derivative data.
+
+### Normalize
```

Comment:
mention here that this step is performed only after the extraction step is completed. So if extract stage in failed normalization won't be started

Reply:
same for load step

Reply:
done